### PR TITLE
src: zlib: revert concatenated-stream changes

### PR DESCRIPTION
**Pull request by chrisdickinson**
This addresses the immediate problem seen in https://github.com/joyent/node/issues/8962. I may try to reintroduce this feature with a flagged option if desired.

### CODE REVIEW

**Comment by misterdjules** [comment]
LGTM. A few comments:
* I'm not sure if using `src: zlib: ` in the first line of the commit message breaks [the contributing rules](http://nodejs.org/documentation/contributing/#index_md_commit). Maybe reword the first line and just mention the sha1 of the reverted commits in the commit message?
* Can we mention issue number #8962 in the commit message?
* How was this tested? Could we ask some people following https://github.com/npm/npm/issues/7043 to test it?

Thanks!

**Comment by othiym23** [comment]
I can confirm that this fixes the crashes that I was seeing when this issue was pointed out to me by @wraithan. @wraithan, does this fix your issues with `node-newrelic` as well?

**Comment by wraithan** [comment]
I'll tat this in a couple hours when I get to work, just saw this email to test this branch

**Comment by wraithan** [comment]
@chrisdickinson @othiym23 This fixes the problem for us. Thanks a ton to both of you for looking into this and fixing it up for me!

**Comment by misterdjules** [comment]
@othiym23 @wraithan @chrisdickinson Thank you very much! Now LGTM without any comment :)

**Comment by chrisdickinson** [comment]
Merged in c8ef97e4d9ef2546d8629c88a9cf4419fc188f22. Thanks all!
